### DISCUSSION TITLE
Skolemnization, formula simplification and pre-processing

### DISCUSSION THREAD

**Discussion by H-Lindner** [bitwuzla/bitwuzla#63] [Q&A] [answered]
Hello,

I am working on adding Bitwuzla to the JavaSMT API and specifically looking at quantifier functionality. In Bitwuzla's C API, `bitwuzla_simplify` has a note that each call to `bitwuzla_check_sat()` simplifies the input formula as a part of pre-processing. I have two questions about this.

1. To which pre-processing options does simplify correspond? Can automatic simplification be disabled?
2. Do either `bitwuzla_simplify` or `BITWUZLA_OPT_PP_NORMALIZE` involve Skolemnization for quantified formulas? 

Thanks!

**Comment by mpreiner** [comment, marked as answer]
Hi @H-Lindner ,

1. `bitwuzla_simplify` calls the preprocessor on all assertions. This step is also done at the beginning of each `bitwuzla_check_sat`
2. No, Skolemization happens lazily during solving.
